corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -289,11 +291,11 @@ def get_case_history(case):
     from corehq.apps.reports.display import xmlns_to_name
 
     changes = defaultdict(dict)
-    for action in case.actions:
-        case_blocks = extract_case_blocks(action.form)
+    for form in FormAccessors(case.domain).get_forms(case.xform_ids, ordered=True):
```

Comment:
I don't think you need to order these here. This will only order the forms in the same order as the IDs that are given. We sort the list of changes later when the function returns. 

Also, this is much more performant than the previous implementation - it fetches all the forms in one go instead of one at a time. Thanks!